src/Umbraco.Core/Persistence/Repositories/Implement/RedirectUrlRepository.cs

```diff
@@ -218,7 +218,21 @@ public IRedirectUrl GetMostRecentUrl(string url, string culture)
                 .OrderByDescending<RedirectUrlDto>(x => x.CreateDateUtc);
             var dtos = Database.Fetch<RedirectUrlDto>(sql);
             var dto = dtos.FirstOrDefault();
+
+            if (dto == null)
+                dto = GetMostRecentUrlWithNoCulture(url);
+
             return dto == null ? null : Map(dto);
         }
+
+        private RedirectUrlDto GetMostRecentUrlWithNoCulture(string url)
+        {
+            var urlHash = url.GenerateHash<SHA1>();
```

Comment:
This is regenerating a hash that was already generated. The url hasn't changed so no need to allocate and add overhead hashing again.

Reply:
This function has now been removed as isn't needed any more :-)